docs/extensibility/annotations.md

```diff
@@ -0,0 +1,69 @@
+# Annotations
+
+Annotations are a way to highlight a specific piece in a Gutenberg post. Examples of this include commenting on a piece of text and spellchecking. Both can use the annotations API to mark a piece of text.
+
+## API
+
+To see the API for yourself the easiest way is to have a block that is at least 200 characters long without formatting and putting the following in the console:
+
+```js
+wp.data.dispatch( 'core/editor' ).addAnnotation( {
+	source: "my-annotations-plugin",
+	blockClientId: wp.data.select( 'core/editor' ).getBlockOrder()[0],
+	range: {
+		startXPath: "text()[1]",
```

Comment:
Hey, thanks for the work on this PR. I don't have all the necessary context to comment properly but here's my concern.

Something is worrying me here: If we're attaching an annotation to a block, why are we using `text` to specify its position? We assume a block have `text` which is not essentially true. Block can have attributes and some attributes can be html.

Can't we rely on attribute names instead of `text` and I don't have strong opinion about `xpath` but this feels like something that should have a relationship with the RichText structure as it already contains ways to target parts of the content in its format API?

Reply:
Block level annotations look good for me though.

Reply:
Could `range: {}` just be e.g. `start: 2, end: 5` so it works with the rich text value? I saw as an argument before that it should be possible to render on the front end. How do you identify rich text field on the front end?

Reply:
Now that we have `richTextIdentifier` I'm thinking we might indeed want to drop the XPaths altogether. My thinking is that when we start working on front-end annotation/commenting API we can persist xpaths, but convert those to the relevant block/positions when retrieving those from the server. I'll change this PR accordingly.